modules/Analytics/components/month-by-month.component.tsx

```diff
@@ -0,0 +1,101 @@
+import dynamic from 'next/dynamic';
+
+// @ts-ignore
+const ResponsiveBarCanvas = dynamic(() => import('@nivo/bar').then(m => m.ResponsiveBarCanvas), { ssr: false });
+
+import { Text, Box as ChackraBox, Tag } from '@chakra-ui/react';
+import { useMemo } from 'react';
+
+import { Box } from '@Components';
+import { IBill } from '@/hooks/useBills';
+import { BillTypes } from '@Modules/Bill/constants/Types';
+
+interface MonthByMonthProps {
+  bills: IBill[];
+}
+
+const keyToText = {
+  rest: 'Retante',
+  income: 'Receita',
+  expense: 'Despesa',
+};
+
+export const MonthByMonth = ({ bills }: MonthByMonthProps) => {
+  const data = useMemo(() => {
+    const billsReduce = bills.reduce((prev, curr) => {
+      const dueDate = new Date(curr.dueDate);
+      const month = `${dueDate.getMonth() + 1}/${dueDate.getFullYear()}`;
+
+      if (!prev[month]) {
+        prev[month] = {};
+
+        if (curr.type === BillTypes.EXPENSE) {
```

Comment:
From what I understand, this first check is to see if there are already values ​​for that specific month/year, if not yet, I put the curr values ​​as default

If my understanding is right

This part is not necessary if you initialize the `expense` and `income` values ​​to 0

Because in the next if you check if `curr.type === billTypes.EXPENSE` and do a `prev[month].expense += curr.value;`

If not, do: `prev[month].income += curr.value;`

Thus, repeating the code

So if the values ​​are initialized with zero, the += operation will work correctly and for cases where the information in the month/year did not yet exist, it becomes the curr value, exactly what the first if does

Reply:
```typescript
 if (!prev[month]) {
        prev[month] = {
          expense: 0,
          income: 0,
          month,
          rest: 0
        };
      }
```

Reply:
But thinking here,
the code checks if there is no value in that `table/year`, if not, it initializes

there depending on whether the value of `curr.type === BillTypes.EXPENSE`
it initializes the expense with the value of `curr.value`
Then, as the value has already been initialized, it checks whether there is a value for the `month/year` (` if (prev?.[month])` )

as the prev for this `month/year` was initialized previously, it does: `prev[month].expense += curr.value;`

so it's as if it did a `curr.values ​​+ curr.values` for cases where the prev didn't yet have a value for that month